lib/utils/stringUtils.js

```diff
@@ -70,11 +70,23 @@ function placeholdEmpty(s) {
 	return s === '' ? '[EMPTY]' : s;
 }
 
+/**
+ * Truncate string to 'limit' length. Append ... if over limit.
+ */
+function truncate(input, limit) {
+	if (input.length > limit) {
+		return input.substring(0, limit) + '...';
```

Comment:
It would be cleaner to use ellipsis instead of 3 dots and substring it to `limit - 1`, so that total length including ellipsis does not exceed the limit.

There is also an edge case when we'll truncate to a space or punctuation character. Somethings like "Some text ..." or "Some text:...".